### PR TITLE
WAVE2 680 Imp publish revisions

### PR DESCRIPTION
**Pull request by nindusaini**

### CODE REVIEW

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core.module [hunk 1]
```diff
@@ -606,6 +606,20 @@ function ucgy_core_panels_pre_render($panels_display, $renderer)
           $fpp->publish_on_scheduled    = $scheduled_to_be_published;
           $fpp->unpublish_on_scheduled  = $scheduled_to_be_unpublished;
 
+        //Now let's get scheduling information for revisions
+        $rev_query = db_select('fpp_scheduling', 'f')
+            ->fields('f', array('fpp_publish', 'fpp_publish_on', 'fpp_publish_on_scheduled', 'revision_id'))
+            ->condition('f.entity_id', $fpp->fpid, '=')
```
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core.module]: Add to these condition lines some inline comments on what is happening to get the revision you want.

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.cron.inc [hunk 2]
```diff
@@ -310,4 +316,51 @@ function get_fpp_page($nid){
...
+    $query->condition('s.fpp_publish_on_scheduled', REQUEST_TIME, '<=');
+    $query_result = $query->execute();
+
+    while ($fpp = $query_result->fetchObject()) {
+        $nid = $fpp->entity_id;
+        $rev_id = $fpp->revision_id;
+        $n = fieldable_panels_panes_load($nid,[$rev_id]);
+        
+        //include the module and inc files so we can call the fpp save function
+        module_load_include('module','fieldable_panels_panes','fieldable_panels_panes');
```
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.cron.inc]: Do you just want a function inside the module file or do you need the entire module loaded? This potentially could be cut down to a single line if you want all functions loaded or just from one include file.

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.cron.inc [hunk 3]
```diff
@@ -310,4 +316,51 @@ function get_fpp_page($nid){
...
+/**
+ * Make a revision current
+ *
+ * @return bool
+ *   TRUE if any revision of an FPP has been updated as current version
+ */
+function _fpp_revisions() {
+
+    $result = FALSE;
+    //module_load_include('module','fieldable_panels_panes','fieldable_panels_panes');
```
Comment (nindusaini) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.cron.inc]: Removed.

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.cron.inc [hunk 4]
```diff
@@ -310,4 +316,51 @@ function get_fpp_page($nid){
...
+
+    // If the time now is greater than the time to make an FPP revision current, make it current it.
```
Comment (nindusaini) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.cron.inc]: Corrected

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.cron.inc (hunk 2 above)
Comment (nindusaini) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.cron.inc]: I just want to use one function in file fpp.module

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.module [hunk 5]
```diff
@@ -22,8 +22,16 @@ function ucgy_core_fpp_scheduling_form_alter(&$form, &$form_state, $form_id)
         //make the revisions tab visible all the time, by default its not visible when an FPP is created only when its updated
         $form['revision']['#access']= true;
         $form['revision']['#title']= 'Revision/Scheduling';
+        $form['revision']['log']['#title']= 'Revision details';
         //Let's make the revisions checkbox un-check by default
-        $form['revision']['revision']['#default_value'] = 0;
+        $form['revision']['revision']['#default_value'] = FALSE;
+        $form['revision']['revision']['#disabled'] = FALSE;
+        unset($form['revision']['revision']['#value']);
```
Comment (nindusaini) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_fpp_scheduling/ucgy_core_fpp_scheduling.module]: FPP API sets a value of this field in some cases, so unsetting it for those cases. I added a comment explaining this.

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core.module (hunk 1 above)
Comment (nindusaini) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core.module]: Comments added.

**Comment by dpfitzsi** [comment]
New code looks good.

**Pull request merged**